File: Java/opentraceability/src/main/java/opentraceability/interfaces/IMasterDataKDE.java
package opentraceability.interfaces;

import opentraceability.utility.XElement;
import org.json.JSONObject;
import org.w3c.dom.Element;



public abstract class IMasterDataKDE {
    public String namespace;
    public String name;
    public Class valueType;

    public abstract void setFromGS1WebVocabJson(Object json);
    public abstract Object getGS1WebVocabJson();
    public abstract void setFromEPCISXml(XElement xml) throws Exception;
    public abstract XElement getEPCISXml() throws Exception;
}

File: Java/OpenTraceability/src/main/kotlin/opentraceability/interfaces/imasterdatakde.kt
package opentraceability.interfaces


import org.w3c.dom.Element
import org.json.JSONObject
import kotlin.reflect.KType

interface IMasterDataKDE {

    var namespace: String
    var name: String
    var valueType: KType

    fun setFromGS1WebVocabJson(json: JSONObject)
    fun getGS1WebVocabJson(): JSONObject?
    fun setFromEPCISXml(xml: Element)
    fun getEPCISXml(): Element?
}
